kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/api/common/NamespacedRef.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.api.common;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import static io.kroxylicious.kubernetes.operator.ResourcesUtil.slug;
+
+/**
+ * Abstraction for references in one kubernetes resource to some kubernetes resource in a specific namespace.
+ * Two LocalRefs are equal iff they have the same group, kind, name and namespace
+ */
+public class NamespacedRef<T> implements Ref<T> {
```

Comment:
Can this be a `record`?

Reply:
If we do that it would be convenient if we went from `getKind()` to `kind()` etc on the Ref interface so that we can have a record that implements Ref more easily.

I'd move from having a delegate field to pulling `group`, `kind`, `name` into fields of the record.

Reply:
done, though I've left renaming the methods of the `ref` interface for now to keep the changes a bit smaller.